### PR TITLE
[APIAP] Update Proxy public/staging urls from Integration -> Settings

### PR DESCRIPTION
**Pull request by Martouta**
Closes [THREESCALE-3576](https://issues.jboss.org/browse/THREESCALE-3576)

### Verification Steps
1. In `config/rolling_updates.yml` you will need `api_as_product: true`
2. In `config/settings.yml` you will need `apicast_custom_url: http://example.com.lvh.me` for this:
https://github.com/3scale/porta/blob/17d259fa26abe0bbbfc710c869cc3bf7c3a26470/app/helpers/api/integrations_helper.rb#L71-L75
3. Run the server with `UNICORN_WORKERS=8 bundle exec rails server -b 0.0.0.0`
4. Through the browser, go to `http://provider-admin.example.com.lvh.me:3000/apiconfig/services/:service_id/settings`
image
5. Change that data and update :dancer:

### CODE REVIEW

File: app/controllers/api/services_controller.rb [hunk 1]
```diff
@@ -75,24 +75,43 @@ def destroy
 
   private
 
+  attr_reader :service
+
+  def integration_settings_result
+    ApiIntegration::SettingsResult.new(service: service, proxy: service.proxy)
+  end
+
+  def create_params
```
Comment (Martouta) [inline app/controllers/api/services_controller.rb]: I know that this is very bad but I encountered conflicts with this during this same morning :scream: And I don't have time to fix it properly for today.

Comment (hallelujah) [reply app/controllers/api/services_controller.rb]: Ah you mean repeating it :)

Yes you can DRY with `user_attribute_names` later

**Comment by thomasmaas** [comment]
@Martouta it allows you to change the url and then it keeps that url when you switch to 3scale managed apicast. That's not the correct behaviour.

**Comment by Martouta** [comment]
Right, @thomasmaas , good catch! :+1: Fixed now :+1: :muscle:

File: app/controllers/api/services_controller.rb (hunk 1 above)
Comment (Martouta) [reply app/controllers/api/services_controller.rb]: In fact it should be done properly in [THREESCALE-3363](https://issues.jboss.org/browse/THREESCALE-3363), but that is after 2.7

File: app/views/api/integrations/apicast/shared/_proxy_endpoints.html.slim [hunk 2]
```diff
@@ -1,4 +1,5 @@
-= f.inputs "API gateway" do
+- is_service_mesh = proxy.service_mesh_integration?
+= f.inputs "API gateway", class: "apicast-only-settings #{'hidden' if is_service_mesh}", disabled: is_service_mesh do
```
Comment (Martouta) [inline app/views/api/integrations/apicast/shared/_proxy_endpoints.html.slim]: I've followed @didierofrivia 's work:
https://github.com/3scale/porta/blob/4f3b989575b030b601c29012fa1ae30d43c8e7b2/app/views/api/services/forms/_authentication_settings.html.slim#L20
https://github.com/3scale/porta/blob/4f3b989575b030b601c29012fa1ae30d43c8e7b2/app/views/api/services/forms/_integration_settings_apiap.html.slim#L1
image
https://github.com/3scale/porta/blob/4f3b989575b030b601c29012fa1ae30d43c8e7b2/app/javascript/packs/settings.js#L4
https://github.com/3scale/porta/blob/4f3b989575b030b601c29012fa1ae30d43c8e7b2/app/javascript/src/Settings/authentication-widget.jsx#L3-L38

File: app/lib/api_integration/settings_result.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    errors
+  end
+
+  private
+
+  # ActiveModel::Errors needs this method to read the errors correctly
+  def read_attribute_for_validation(attr)
+    public_send(attr)
+  end
+end
```
Comment (duduribeiro) [inline app/lib/api_integration/settings_result.rb]: why to put this in `lib` ? 🤔 

I think it is better to put it in `app/models` or `app/services`.. what do you think?

**Pull request merged**